playground/backend/internal/utils/code_processing_utils_test.go

```diff
@@ -0,0 +1,430 @@
+			expectedStatus:        pb.Status_STATUS_FINISHED,
+			expectedCompileOutput: "",
+			expectedRunOutput:     "Hello world!\n",
+			expectedRunError:      nil,
+			args: args{
+				ctx:        context.Background(),
+				env:        env,
+				sdk:        pb.Sdk_SDK_JAVA,
+				pipelineId: uuid.New(),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lc, _ := fs_tool.NewLifeCycle(pb.Sdk_SDK_JAVA, tt.args.pipelineId, os.Getenv("APP_WORK_DIR"))
+			filePath := lc.GetAbsoluteExecutableFilePath()
+			workingDir := lc.GetAbsoluteExecutableFilesFolderPath()
+
+			exec := executors.NewExecutorBuilder().
+				WithValidator().
+				WithSdkValidators(validators.GetJavaValidators(filePath)).
+				WithCompiler().
+				WithCommand(sdkEnv.ExecutorConfig.CompileCmd).
+				WithArgs(sdkEnv.ExecutorConfig.CompileArgs).
+				WithFileName(filePath).
+				WithWorkingDir(workingDir)
+
+			err := lc.CreateFolders()
+			if err != nil {
+				t.Fatalf("error during prepare folders: %s", err.Error())
+			}
+			if tt.createExecFile {
+				_, _ = lc.CreateExecutableFile(tt.code)
+			}
+
+			if tt.cancelFunc {
+				go func(ctx context.Context, pipelineId uuid.UUID) {
+					// to imitate behavior of cancellation
+					time.Sleep(5 * time.Second)
+					cacheService.SetValue(ctx, pipelineId, cache.Canceled, true)
+				}(tt.args.ctx, tt.args.pipelineId)
+			}
+			Process(tt.args.ctx, cacheService, lc, exec, tt.args.pipelineId, tt.args.env)
+
+			status, _ := cacheService.GetValue(tt.args.ctx, tt.args.pipelineId, cache.Status)
+			if !reflect.DeepEqual(status, tt.expectedStatus) {
+				t.Errorf("processCode() set status: %s, but expectes: %s", status, tt.expectedStatus)
+			}
+
+			compileOutput, _ := cacheService.GetValue(tt.args.ctx, tt.args.pipelineId, cache.CompileOutput)
+			if tt.expectedCompileOutput != nil && strings.Contains(tt.expectedCompileOutput.(string), "%s") {
+				tt.expectedCompileOutput = fmt.Sprintf(tt.expectedCompileOutput.(string), lc.GetAbsoluteExecutableFilePath())
+			}
+			if !reflect.DeepEqual(compileOutput, tt.expectedCompileOutput) {
+				t.Errorf("processCode() set compileOutput: %s, but expectes: %s", compileOutput, tt.expectedCompileOutput)
+			}
+
+			runOutput, _ := cacheService.GetValue(tt.args.ctx, tt.args.pipelineId, cache.RunOutput)
+			if !reflect.DeepEqual(runOutput, tt.expectedRunOutput) {
+				t.Errorf("processCode() set runOutput: %s, but expectes: %s", runOutput, tt.expectedRunOutput)
+			}
+
+			runError, _ := cacheService.GetValue(tt.args.ctx, tt.args.pipelineId, cache.RunError)
+			if tt.expectedRunError != nil && strings.Contains(tt.expectedRunError.(string), "%s") {
+				tt.expectedRunError = fmt.Sprintf(tt.expectedRunError.(string), tt.args.pipelineId)
+			}
+			if !reflect.DeepEqual(runError, tt.expectedRunError) {
+				t.Errorf("processCode() set runError: %s, but expectes: %s", runError, tt.expectedRunError)
+			}
+
+			// remove
+			path := os.Getenv("APP_WORK_DIR") + "/executable_files"
```

Comment:
Don't we have `/executable_files` somewhere in the constants of another file?

Reply:
`java_fs` file contains this constant:
```
const (
	javaBaseFileFolder          = "executable_files"
	...
)
```
but it is private.